operator/v1/types_ingress.go

```diff
@@ -351,6 +351,10 @@ type IngressControllerStatus struct {
 	// tlsProfile is the TLS connection configuration that is in effect.
 	// +optional
 	TLSProfile *configv1.TLSProfileSpec `json:"tlsProfile,omitempty"`
+
+	// observedGeneration is the most recent generation observed.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
```

Comment:
The change itself looks good, but you're getting closer to the generic `OperatorStatus` have you considered switching to that?